apps/user/tests/test_schemas.py

```diff
@@ -563,3 +564,88 @@ def _query_check(filters, **kwargs):
+                    results {
+                    id
+                    emailDisplay
+                    firstName
+                    isActive
+                    language
+                    lastName
+                    displayName
+                    displayPictureUrl
+                    organization
+                    }
+                    totalCount
+                }
+            }
+
+        '''
+        user = UserFactory.create(email='testuser@deep.com')
+        UserFactory.create(email='testuser2@deep.com')
+        UserFactory.create(email='testuser3@deep.com')
+        # # Without Login session
+        self.query_check(query_single_user, variables={'id': str(user.id)}, assert_for_error=True)
+
+        # # Login
+        self.force_login(user)
+
+        # Query User (Success)
+        content = self.query_check(query_single_user, variables={'id': str(user.id)})
+        self.assertEqual(content['data']['user']['id'], str(user.id), content)
+        self.assertEqual(content['data']['user']['emailDisplay'], 'te...er@deep.com')
+
+        # Query Users (Success)
+        content = self.query_check(query_all_users)
+        email_display_list = [result['emailDisplay'] for result in content['data']['users']['results']]
+        self.assertTrue(set(['te...er@deep.com', 'te...r2@deep.com']).issubset(set(email_display_list)))
+
+    def test_generate_hidden_email(self):
+        user1 = UserFactory.create(email='testuser1@deep.com')
+        user2 = UserFactory.create(email='testuser2@deep.com')
+        user3 = UserFactory.create(email='abcd@deep.com')
+        user4 = UserFactory.create(email='abc@deep.com')
+        user5 = UserFactory.create(email='xy@deep.com')
+        user6 = UserFactory.create(email='a@deep.com')
+
+        user1_expected_hidden_email = 'te...r1@deep.com'
+        user2_expected_hidden_email = 'te...r2@deep.com'
+        user3_expected_hidden_email = 'a...d@deep.com'
+        user4_expected_hidden_email = 'a...c@deep.com'
+        user5_expected_hidden_email = 'x...y@deep.com'
+        user6_expected_hidden_email = 'a...a@deep.com'
+
+        user1_generated_email = generate_hidden_email(user1.email)
+        user2_generated_email = generate_hidden_email(user2.email)
+        user3_generated_email = generate_hidden_email(user3.email)
+        user4_generated_email = generate_hidden_email(user4.email)
+        user5_generated_email = generate_hidden_email(user5.email)
+        user6_generated_email = generate_hidden_email(user6.email)
+
+        self.assertEqual(user1_expected_hidden_email, user1_generated_email)
+        self.assertEqual(user2_expected_hidden_email, user2_generated_email)
+        self.assertEqual(user3_expected_hidden_email, user3_generated_email)
+        self.assertEqual(user4_expected_hidden_email, user4_generated_email)
+        self.assertEqual(user5_expected_hidden_email, user5_generated_email)
+        self.assertEqual(user6_expected_hidden_email, user6_generated_email)
```

Comment:
@Rup-Narayan-Rajbanshi Let's only test the function, UserFactory is not required here.

```suggestion
        for original, expected in [
            ('testuser1@deep.com', 'te...r1@deep.com'),
            ('testuser2@deep.com', 'te...r2@deep.com'),
            ('abcd@deep.com', 'a...d@deep.com'),
            ('abc@deep.com', 'a...c@deep.com'
            ('xy@deep.com', 'x...y@deep.com'),
            ('a@deep.com', 'a...a@deep.com'),
        ]:
            self.assertEqual(expected, generate_hidden_email(original))
```